src/main/kotlin/org/rust/lang/core/types/Extensions.kt

```diff
@@ -27,7 +28,13 @@ val RsTypeReference.type: Ty
 val RsTypeElement.lifetimeElidable: Boolean
     get() {
         val typeOwner = owner.parent
-        return typeOwner !is RsFieldDecl && typeOwner !is RsTupleFieldDecl && typeOwner !is RsTypeAlias
+
+        val isAssociatedConstant =  typeOwner is RsConstant
+            && (typeOwner.owner is RsAbstractableOwner.Trait
```

Comment:
Let's write something like the following instead
`val isAssociatedConstant = typeOwner is RsConstant && typeOwner.owner.isImplOrTrait`
It looks a little more readable